packages/relay/src/lib/eth.ts

```diff
@@ -568,34 +569,35 @@ export class EthImpl implements Eth {
    * @param hash
    */
   async getTransactionReceipt(hash: string) {
-    // FIXME: This should go to the mirror node. For now we have an in memory map we
-    //        grab receipts from for each tx that was executed. Even in the full implementation
-    //        we will want to use the cache as an LRU to avoid hammering the mirror node more
-    //        than needed, since the receipt is an immutable type. But the cache should be in the
-    //        mirror node connector, rather than here.
-    this.logger.trace('getTransactionReceipt(hash=%s)', hash);
-    try {
-      // Lookup the receipt in the cache.
-      const receipt = cache.get(hash);
-
-      if (!receipt) {
-        // FIXME: if it wasn't in the cache, go to the mirror node. If it isn't there either,
-        //        then return null.
-        this.logger.debug(
-          'Did not find the receipt for hash %s. Returning null.',
-          hash
-        );
-        return null;
-      }
-
-      this.logger.trace('Found a receipt, returning it to the caller');
-      return receipt;
-    } catch (e) {
-      this.logger.error(
-        e,
-        'Failed to handle getTransactionReceipt cleanly for hash %s',
-        hash
-      );
+    this.logger.trace(`getTransactionReceipt(${hash})`);
+    const receiptResponse = await this.mirrorNodeClient.getContractResult(hash);
+    this.logger.trace(`response - ${JSON.stringify(receiptResponse)}`);
+    if (receiptResponse === null || receiptResponse.hash === undefined) {
+      this.logger.trace(`no receipt for ${hash}`);
+      // block not found
+      return Promise.reject('no receipt');
+    } else {
+      const effectiveGas =
+        receiptResponse.max_fee_per_gas === undefined ||
+        receiptResponse.max_fee_per_gas == '0x'
+          ? receiptResponse.gas_price
+          : receiptResponse.max_fee_per_gas;
+      const answer = {
+        blockHash: receiptResponse.block_hash.substring(0, 66),
+        blockNumber: EthImpl.prepend0x(receiptResponse.block_number.toString(16)),
+        from: receiptResponse.from,
+        cumulativeGasUsed: EthImpl.prepend0x(receiptResponse.block_gas_used.toString(16)),
+        gasUsed: EthImpl.prepend0x(receiptResponse.gas_used.toString(16)),
+        logs: receiptResponse.logs,
+        logsBloom: receiptResponse.logs_bloom,
+        transactionHash: receiptResponse.hash,
+        transactionIndex: EthImpl.prepend0x(receiptResponse.transaction_index.toString(16)),
+        effectiveGasPrice: EthImpl.prepend0x((Number.parseInt(effectiveGas) * 10_000_000_000).toString(16)),
+        contractAddress: receiptResponse.to,
+        status: receiptResponse.status,
```

Comment:
I don't think such a value exist on the response. This is supposed to be 0 or 1 right?

Reply:
It exists on the mirror node response.  0x0 or 0x1 as appropriate

Reply:
Ah you're right.
Open API Spec wasn't ordered so i forgot it was in there